From 53cbba496bd9379a83a8aa3e123295c97887debd Mon Sep 17 00:00:00 2001
From: Sam Macbeth <sam@cliqz.com>
Date: Mon, 24 Aug 2020 16:19:12 +0200
Subject: Ghostery Logo Added

---
 app/build.gradle | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/app/build.gradle b/app/build.gradle
index f346771cc..fe68d68f4 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -34,7 +34,7 @@ android {
     }
 
     defaultConfig {
-        applicationId "org.mozilla"
+        applicationId "com.ghostery.android"
         minSdkVersion Config.minSdkVersion
         targetSdkVersion Config.targetSdkVersion
         versionCode 1
@@ -147,6 +147,23 @@ android {
                     "deepLinkScheme": deepLinkSchemeValue
             ]
         }
+        ghostery releaseTemplate >> {
+            buildConfigField "boolean", "USE_RELEASE_VERSIONING", "true"
+            applicationIdSuffix ".ghostery"
+            def deepLinkSchemeValue = "ghostery"
+            buildConfigField "String", "DEEP_LINK_SCHEME", "\"$deepLinkSchemeValue\""
+            manifestPlaceholders = [
+                    // This release type is meant to replace Firefox (Release channel) and therefore needs to inherit
+                    // its sharedUserId for all eternity. See:
+                    // https://searchfox.org/mozilla-central/search?q=moz_android_shared_id&case=false&regexp=false&path=
+                    // Shipping an app update without sharedUserId can have
+                    // fatal consequences. For example see:
+                    //  - https://issuetracker.google.com/issues/36924841
+                    //  - https://issuetracker.google.com/issues/36905922
+                    "sharedUserId": "com.ghostery.android.ghostery.sharedID",
+                    "deepLinkScheme": deepLinkSchemeValue
+            ]
+        }
     }
 
     buildFeatures {
-- 
2.37.2

