From 14bb526264b3c6a93151e5a213be6ee2c608ab76 Mon Sep 17 00:00:00 2001
From: Stefano Pacifici <stefano.pacifici@gmail.com>
Date: Tue, 9 Jun 2020 10:56:17 +0200
Subject: Removing FirefoxSync settings

---
 .../fenix/settings/SettingsFragment.kt        | 21 ++++++++++---------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt b/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
index 1a1a2cc2f..153948648 100644
--- a/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
+++ b/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
@@ -13,7 +13,6 @@ import android.os.Bundle
 import android.os.Handler
 import android.provider.Settings
 import android.widget.Toast
-import androidx.appcompat.content.res.AppCompatResources
 import androidx.lifecycle.lifecycleScope
 import androidx.navigation.NavDirections
 import androidx.navigation.findNavController
@@ -22,7 +21,6 @@ import androidx.preference.PreferenceCategory
 import androidx.preference.PreferenceFragmentCompat
 import androidx.recyclerview.widget.RecyclerView
 import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers.Main
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 import mozilla.components.concept.sync.AccountObserver
@@ -34,14 +32,7 @@ import org.mozilla.fenix.Config
 import org.mozilla.fenix.HomeActivity
 import org.mozilla.fenix.R
 import org.mozilla.fenix.components.metrics.Event
-import org.mozilla.fenix.ext.application
-import org.mozilla.fenix.ext.components
-import org.mozilla.fenix.ext.getPreferenceKey
-import org.mozilla.fenix.ext.metrics
-import org.mozilla.fenix.ext.requireComponents
-import org.mozilla.fenix.ext.settings
-import org.mozilla.fenix.ext.showToolbar
-import org.mozilla.fenix.ext.toRoundedDrawable
+import org.mozilla.fenix.ext.*
 import org.mozilla.fenix.settings.account.AccountAuthErrorPreference
 import org.mozilla.fenix.settings.account.AccountPreference
 import kotlin.system.exitProcess
@@ -382,6 +373,8 @@ class SettingsFragment : PreferenceFragmentCompat() {
      * Updates the UI to reflect current account state.
      * Possible conditions are logged-in without problems, logged-out, and logged-in but needs to re-authenticate.
      */
+    /* Ghostery Begin: disabling FirefoxSync entry points **/
+    @Suppress("UNUSED_PARAMETER")
     private fun updateAccountUIState(context: Context, profile: Profile?) {
         val preferenceSignIn =
             requirePreference<Preference>(R.string.pref_key_sign_in)
@@ -392,6 +385,12 @@ class SettingsFragment : PreferenceFragmentCompat() {
         val accountPreferenceCategory =
             requirePreference<PreferenceCategory>(R.string.pref_key_account_category)
 
+        preferenceSignIn.isVisible = false
+        preferenceFirefoxAccount.isVisible = false
+        preferenceFirefoxAccountAuthError.isVisible = false
+        accountPreferenceCategory.isVisible = false
+
+        /*
         val accountManager = requireComponents.backgroundServices.accountManager
         val account = accountManager.authenticatedAccount()
 
@@ -438,6 +437,8 @@ class SettingsFragment : PreferenceFragmentCompat() {
             preferenceFirefoxAccountAuthError.isVisible = false
             accountPreferenceCategory.isVisible = false
         }
+        */
+        /* Ghostery End */
     }
 
     private fun updateFxASyncOverrideMenu() {
-- 
2.25.1

