From 05e87fe685eb5016a42f3e9b8a1e14285b16b860 Mon Sep 17 00:00:00 2001
From: Sam Macbeth <sam@macbeth.cc>
Date: Tue, 4 Aug 2020 10:31:31 +0200
Subject: Remove addons from menu

---
 .../components/toolbar/DefaultToolbarMenu.kt   | 18 +++++-------------
 1 file changed, 5 insertions(+), 13 deletions(-)

diff --git a/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt b/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
index 3a51f694d..5c6837573 100644
--- a/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
+++ b/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
@@ -15,6 +15,7 @@ import kotlinx.coroutines.Job
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.flow.mapNotNull
 import kotlinx.coroutines.launch
+import mozilla.components.browser.menu.BrowserMenuBuilder
 import mozilla.components.browser.menu.BrowserMenuHighlight
 import mozilla.components.browser.menu.WebExtensionBrowserMenuBuilder
 import mozilla.components.browser.menu.item.BrowserMenuDivider
@@ -76,19 +77,10 @@ open class DefaultToolbarMenu(
         get() = store.state.selectedTab
 
     override val menuBuilder by lazy {
-        WebExtensionBrowserMenuBuilder(
-            items = coreMenuItems,
-            endOfMenuAlwaysVisible = shouldUseBottomToolbar,
-            store = store,
-            style = WebExtensionBrowserMenuBuilder.Style(
-                webExtIconTintColorResource = primaryTextColor(),
-                addonsManagerMenuItemDrawableRes = R.drawable.ic_addons_extensions
-            ),
-            onAddonsManagerTapped = {
-                onItemTapped.invoke(ToolbarMenu.Item.AddonsManager)
-            },
-            appendExtensionSubMenuAtStart = shouldUseBottomToolbar
-        )
+       BrowserMenuBuilder(
+           coreMenuItems,
+           endOfMenuAlwaysVisible = shouldUseBottomToolbar
+       )
     }
 
     override val menuToolbar by lazy {
-- 
2.35.1

