From 7aa9b342cd43fc7835213dc366cf80ea87eda395 Mon Sep 17 00:00:00 2001
From: Sam Macbeth <sam@macbeth.cc>
Date: Tue, 4 Aug 2020 10:31:31 +0200
Subject: Remove addons from menu

---
 .../components/toolbar/DefaultToolbarMenu.kt     | 16 ++++------------
 1 file changed, 4 insertions(+), 12 deletions(-)

diff --git a/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt b/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
index 799a55ea58..94b8d6f416 100644
--- a/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
+++ b/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
@@ -14,6 +14,7 @@ import androidx.lifecycle.lifecycleScope
 import kotlinx.coroutines.Job
 import kotlinx.coroutines.flow.mapNotNull
 import kotlinx.coroutines.launch
+import mozilla.components.browser.menu.BrowserMenuBuilder
 import mozilla.components.browser.menu.BrowserMenuHighlight
 import mozilla.components.browser.menu.BrowserMenuItem
 import mozilla.components.browser.menu.WebExtensionBrowserMenuBuilder
@@ -79,19 +80,9 @@ open class DefaultToolbarMenu(
         get() = store.state.selectedTab

     override val menuBuilder by lazy {
-        FxNimbus.features.print.recordExposure()
-        WebExtensionBrowserMenuBuilder(
-            items = coreMenuItems,
-            endOfMenuAlwaysVisible = shouldUseBottomToolbar,
-            store = store,
-            style = WebExtensionBrowserMenuBuilder.Style(
-                webExtIconTintColorResource = primaryTextColor(),
-                addonsManagerMenuItemDrawableRes = R.drawable.ic_addons_extensions,
-            ),
-            onAddonsManagerTapped = {
-                onItemTapped.invoke(ToolbarMenu.Item.AddonsManager)
-            },
-            appendExtensionSubMenuAtStart = shouldUseBottomToolbar,
+        BrowserMenuBuilder(
+            coreMenuItems,
+            endOfMenuAlwaysVisible = shouldUseBottomToolbar
         )
     }

--
2.39.2 (Apple Git-143)

