From c4a67cc287a1b259a7b0d1f400c3e626e84447d4 Mon Sep 17 00:00:00 2001
From: Krzysztof Modras <chrmod@chrmod.net>
Date: Mon, 28 Feb 2022 16:54:55 +0100
Subject: Search widget: replace logo

---
 .../gecko/search/SearchWidgetProvider.kt      | 24 ++++++++-----------
 1 file changed, 10 insertions(+), 14 deletions(-)

diff --git a/fenix/app/src/main/java/org/mozilla/gecko/search/SearchWidgetProvider.kt b/fenix/app/src/main/java/org/mozilla/gecko/search/SearchWidgetProvider.kt
index 4656b04d57..4fe5456847 100644
--- a/fenix/app/src/main/java/org/mozilla/gecko/search/SearchWidgetProvider.kt
+++ b/fenix/app/src/main/java/org/mozilla/gecko/search/SearchWidgetProvider.kt
@@ -11,6 +11,7 @@ import android.appwidget.AppWidgetProvider
 import android.content.ComponentName
 import android.content.Context
 import android.content.Intent
+import android.graphics.Color
 import android.os.Build
 import android.os.Bundle
 import android.speech.RecognizerIntent
@@ -165,20 +166,15 @@ class SearchWidgetProvider : AppWidgetProvider() {
 
     private fun RemoteViews.setIcon(context: Context) {
         // gradient color available for android:fillColor only on SDK 24+
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-            setImageViewResource(
-                R.id.button_search_widget_new_tab_icon,
-                R.drawable.ic_launcher_foreground,
-            )
-        } else {
-            setImageViewBitmap(
-                R.id.button_search_widget_new_tab_icon,
-                AppCompatResources.getDrawable(
-                    context,
-                    R.drawable.ic_launcher_foreground,
-                )?.toBitmap(),
-            )
-        }
+        setImageViewBitmap(
+            R.id.button_search_widget_new_tab_icon,
+            AppCompatResources.getDrawable(
+                context,
+                R.drawable.ic_launcher_foreground
+            )?.mutate()?.apply {
+                setTint(Color.rgb(0x00, 0xAE, 0xF0))
+            }?.toBitmap()
+        )
 
         val appName = context.getString(R.string.app_name)
         setContentDescription(
-- 
2.39.2 (Apple Git-143)

