From 34aae1d29b94dd4b83df303915a65478001e2ded Mon Sep 17 00:00:00 2001
From: Sam Macbeth <sam@macbeth.cc>
Date: Tue, 4 Aug 2020 10:31:31 +0200
Subject: [PATCH] Remove addons from menu

---
 .../components/toolbar/DefaultToolbarMenu.kt   | 18 +++++-------------
 1 file changed, 5 insertions(+), 13 deletions(-)

diff --git a/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt b/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
index 6dfe58dca..b5c88b26e 100644
--- a/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
+++ b/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
@@ -15,6 +15,7 @@ import kotlinx.coroutines.Job
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.flow.mapNotNull
 import kotlinx.coroutines.launch
+import mozilla.components.browser.menu.BrowserMenuBuilder
 import mozilla.components.browser.menu.BrowserMenuHighlight
 import mozilla.components.browser.menu.WebExtensionBrowserMenuBuilder
 import mozilla.components.browser.menu.item.BrowserMenuDivider
@@ -74,19 +75,10 @@ open class DefaultToolbarMenu(
         get() = store.state.selectedTab
 
     override val menuBuilder by lazy {
-        WebExtensionBrowserMenuBuilder(
-            items = coreMenuItems,
-            endOfMenuAlwaysVisible = shouldUseBottomToolbar,
-            store = store,
-            style = WebExtensionBrowserMenuBuilder.Style(
-                webExtIconTintColorResource = primaryTextColor(),
-                addonsManagerMenuItemDrawableRes = R.drawable.ic_addons_extensions
-            ),
-            onAddonsManagerTapped = {
-                onItemTapped.invoke(ToolbarMenu.Item.AddonsManager)
-            },
-            appendExtensionSubMenuAtStart = shouldUseBottomToolbar
-        )
+       BrowserMenuBuilder(
+           coreMenuItems,
+           endOfMenuAlwaysVisible = shouldUseBottomToolbar
+       )
     }
 
     override val menuToolbar by lazy {
-- 
2.34.1

