From 506a6bd8d12b304d4345564e82f84f173d631e95 Mon Sep 17 00:00:00 2001
From: Sam Macbeth <sam@macbeth.cc>
Date: Tue, 4 Aug 2020 10:31:31 +0200
Subject: Remove addons from menu

---
 .../fenix/components/toolbar/DefaultToolbarMenu.kt        | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt b/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
index 9838a622c..f9488404f 100644
--- a/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
+++ b/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
@@ -11,6 +11,7 @@ import androidx.lifecycle.LifecycleOwner
 import androidx.lifecycle.lifecycleScope
 import kotlinx.coroutines.Job
 import kotlinx.coroutines.launch
+import mozilla.components.browser.menu.BrowserMenuBuilder
 import mozilla.components.browser.menu.BrowserMenuHighlight
 import mozilla.components.browser.menu.item.BrowserMenuDivider
 import mozilla.components.browser.menu.item.BrowserMenuHighlightableItem
@@ -61,6 +62,12 @@ class DefaultToolbarMenu(
     private val session: Session? get() = sessionManager.selectedSession
 
     override val menuBuilder by lazy {
+        /* Ghostery Begin: removing add-ons from the default menu */
+        BrowserMenuBuilder(
+            menuItems,
+            endOfMenuAlwaysVisible = !shouldReverseItems
+        )
+        /* Ghostery: no web extensions +/
         WebExtensionBrowserMenuBuilder(
             menuItems,
             endOfMenuAlwaysVisible = !shouldReverseItems,
@@ -71,6 +78,7 @@ class DefaultToolbarMenu(
             },
             appendExtensionSubMenuAtStart = !shouldReverseItems
         )
+         /+ Ghostery End */
     }
 
     override val menuToolbar by lazy {
-- 
2.25.1

