From 1ad2022cfd8ea22f9b01df6e3f889d37ef457aa2 Mon Sep 17 00:00:00 2001
From: Sam Macbeth <sam@macbeth.cc>
Date: Tue, 4 Aug 2020 10:31:31 +0200
Subject: Remove addons from menu

---
 .../fenix/components/toolbar/DefaultToolbarMenu.kt        | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt b/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
index 214a20974..7b1ef61c2 100644
--- a/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
+++ b/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt
@@ -11,6 +11,7 @@ import androidx.lifecycle.LifecycleOwner
 import androidx.lifecycle.lifecycleScope
 import kotlinx.coroutines.Job
 import kotlinx.coroutines.launch
+import mozilla.components.browser.menu.BrowserMenuBuilder
 import mozilla.components.browser.menu.BrowserMenuHighlight
 import mozilla.components.browser.menu.WebExtensionBrowserMenuBuilder
 import mozilla.components.browser.menu.item.BrowserMenuDivider
@@ -60,6 +61,12 @@ class DefaultToolbarMenu(
     private val session: Session? get() = sessionManager.selectedSession
 
     override val menuBuilder by lazy {
+        /* Ghostery Begin: removing add-ons from the default menu */
+        BrowserMenuBuilder(
+            menuItems,
+            endOfMenuAlwaysVisible = !shouldReverseItems
+        )
+        /* Ghostery: no web extensions +/
         WebExtensionBrowserMenuBuilder(
             menuItems,
             endOfMenuAlwaysVisible = !shouldReverseItems,
@@ -70,6 +77,7 @@ class DefaultToolbarMenu(
             },
             appendExtensionSubMenuAtStart = !shouldReverseItems
         )
+         /+ Ghostery End */
     }
 
     override val menuToolbar by lazy {
-- 
2.28.0

