From 358f992839a6acb17a34b6a12b57d365dfcd744c Mon Sep 17 00:00:00 2001
From: Sam Macbeth <sam@cliqz.com>
Date: Sat, 19 Sep 2020 18:56:25 +0200
Subject: Fix signing config

---
 app/build.gradle | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/app/build.gradle b/app/build.gradle
index 6c7411afd..ab66fc0b4 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -56,13 +56,12 @@ android {
 
         if (gradle.hasProperty("localProperties.autosignReleaseWithDebugKey")) {
             signingConfig signingConfigs.debug
-        } else {
-            signingConfig signingConfigs.release
         }
 
         if (gradle.hasProperty("localProperties.debuggable")) {
             debuggable true
         }
+        signingConfig signingConfigs.release
     }
 
     buildTypes {
-- 
2.28.0

