From 111779630d167d2c8dff2cfed9fc0a01c6e3d773 Mon Sep 17 00:00:00 2001
From: Sam Macbeth <sam@cliqz.com>
Date: Fri, 13 Nov 2020 15:53:45 +0100
Subject: Reenable sentry in build gradle

---
 app/build.gradle | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/app/build.gradle b/app/build.gradle
index 2f7eec04a..bd7360878 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -237,8 +237,19 @@ android.applicationVariants.all { variant ->
 // -------------------------------------------------------------------------------------------------
 
     buildConfigField 'String', 'SENTRY_TOKEN', 'null'
+    if (!isDebug) {
+        // Reading sentry token from local file (if it exists). In a release task on taskcluster it will be available.
+        try {
+            def token = new File("${rootDir}/.sentry_token").text.trim()
+            buildConfigField 'String', 'SENTRY_TOKEN', '"' + token + '"'
+        } catch (FileNotFoundException ignored) {}
+    }
     buildConfigField 'boolean', 'CRASH_REPORTING', 'false'
-    buildConfigField 'boolean', 'TELEMETRY', 'true'
+    if (!isDebug) {
+        buildConfigField 'boolean', 'TELEMETRY', 'true'
+    } else {
+        buildConfigField 'boolean', 'TELEMETRY', 'false'
+    }
 
     def buildDate = Config.generateBuildDate()
     // Setting buildDate with every build changes the generated BuildConfig, which slows down the
-- 
2.29.2

