From a7d320424c83710dca45af19939efdfa5c541d6f Mon Sep 17 00:00:00 2001
From: Sam Macbeth <sam@cliqz.com>
Date: Fri, 13 Nov 2020 17:52:35 +0100
Subject: Remove glean plugin

---
 app/build.gradle | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/app/build.gradle b/app/build.gradle
index bd7360878..09f143d14 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -319,11 +319,6 @@ androidExtensions {
     experimental = true
 }
 
-// Generate Kotlin code and markdown docs for the Fenix Glean metrics.
-ext.gleanGenerateMarkdownDocs = true
-ext.gleanDocsDirectory = "$rootDir/docs"
-apply plugin: "org.mozilla.telemetry.glean-gradle-plugin"
-
 dependencies {
     debugImplementation Deps.mozilla_browser_engine_gecko_nightly
 
@@ -497,7 +492,7 @@ dependencies {
 
     // For the initial release of Glean 19, we require consumer applications to
     // depend on a separate library for unit tests. This will be removed in future releases.
-    testImplementation "org.mozilla.telemetry:glean-forUnitTests:${project.ext.glean_version}"
+    // testImplementation "org.mozilla.telemetry:glean-forUnitTests:${project.ext.glean_version}"
 
     lintChecks project(":mozilla-lint-rules")
 }
-- 
2.29.2

